File: web-automation/src/test/java/com/cholemetric/web/pages/SignUpPage.java
package com.cholemetric.web.pages;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class SignUpPage extends BasePage {
    private static final By FULL_NAME = By.id("fullName");
    private static final By EMAIL = By.id("email");
    private static final By PASSWORD = By.id("password");
    private static final By CONFIRM_PASSWORD = By.id("confirmPassword");
    private static final By HOSPITAL = By.id("hospital");
    private static final By SPECIALIZATION = By.id("specialization");
    private static final By SIGN_UP_BUTTON = By.cssSelector("button[type='submit'], .btn-signup, #btnSignUp");
    private static final By LOGIN_LINK = By.cssSelector("a[href*='login'], .login-link");
    private static final By ERROR_MESSAGE = By.cssSelector(".error, .alert-danger, #errorMsg");

    public SignUpPage(WebDriver driver) { super(driver); }

    public void navigateTo(String baseUrl) { driver.get(baseUrl + "signup.html"); }
    public void enterFullName(String name) { type(FULL_NAME, name); }
    public void enterEmail(String email) { type(EMAIL, email); }
    public void enterPassword(String password) { type(PASSWORD, password); }
    public void enterConfirmPassword(String pass) { type(CONFIRM_PASSWORD, pass); }
    public void enterHospital(String hospital) { type(HOSPITAL, hospital); }
    public void enterSpecialization(String spec) { type(SPECIALIZATION, spec); }
    public void clickSignUpButton() { click(SIGN_UP_BUTTON); }
    public void clickLoginLink() { click(LOGIN_LINK); }
    public boolean isSignUpPageDisplayed() { return isDisplayed(SIGN_UP_BUTTON); }
    public boolean isErrorMessageDisplayed() { return isDisplayed(ERROR_MESSAGE); }
    public String getErrorMessage() { try { return getText(ERROR_MESSAGE); } catch(Exception e) { return ""; } }
    public void register(String name, String email, String password, String hospital, String spec) {
        enterFullName(name); enterEmail(email); enterPassword(password);
        enterConfirmPassword(password); enterHospital(hospital); enterSpecialization(spec);
        clickSignUpButton();
    }
}

File: automation/selenium_tests/pages/signup_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage

class SignupPage(BasePage):
    """Page Object Model for Cholemetric Signup Page."""
    
    # Locators
    FULLNAME_INPUT = (By.ID, "full_name")
    EMAIL_INPUT = (By.ID, "signup_email")
    MEDICAL_ID_INPUT = (By.ID, "medical_id")
    PASSWORD_INPUT = (By.ID, "signup_password")
    CONFIRM_PASS_INPUT = (By.ID, "confirm_password")
    SIGNUP_BTN = (By.ID, "btn_signup")
    LOGIN_LINK = (By.XPATH, "//a[contains(@href, 'login_form')]")

    def navigate_to_signup(self, base_url):
        self.open(f"{base_url}/signup.html")

    def register_user(self, name, email, medical_id, password):
        self.type(self.FULLNAME_INPUT, name)
        self.type(self.EMAIL_INPUT, email)
        self.type(self.MEDICAL_ID_INPUT, medical_id)
        self.type(self.PASSWORD_INPUT, password)
        self.type(self.CONFIRM_PASS_INPUT, password)
        self.click(self.SIGNUP_BTN)

    def is_signup_page_loaded(self):
        return self.is_displayed(self.FULLNAME_INPUT) and self.is_displayed(self.SIGNUP_BTN)
